client/my-sites/stats/stats-download-csv/index.jsx

```diff
@@ -87,7 +86,7 @@ const connectComponent = connect( ( state, ownProps ) => {
 		isLoading = dataList.isLoading();
 	} else {
 		data = getSiteStatsCSVData( state, siteId, statType, query );
-		isLoading = isRequestingSiteStatsForQuery( state, siteId, statType, query );
+		isLoading = isRequestingSiteStatsForQuery( state, siteId, statType, query ) && ! data.length;
```

Comment:
I'd personally move the `data.length` check for this to the render function since `data` is already in the props & `isLoading` isn't semantically correct if the result is just empty.

Something like [this](https://gist.github.com/anonymous/049d7a0f73b44c5e416d7fb22d0177ac/178fb0438dccba1871cee7bb8d37d1a86434cb77)